extension/src/popup/components/sendPayment/SendConfirm/TransactionDetails/index.tsx

```diff
@@ -166,16 +175,24 @@ export const TransactionDetails = ({
       <div className="TransactionDetails__row">
         <div>Sending to </div>
         <div className="TransactionDetails__row__right">
-          <div className="TransactionDetails__identicon">
-            <IdenticonImg publicKey={destination} />
-            <span>{truncatedPublicKey(destination)}</span>
-          </div>
+          {federationAddress ? (
+            <div>
+              <span>{federationAddress}</span>
+            </div>
+          ) : (
+            <div className="TransactionDetails__identicon">
+              <IdenticonImg publicKey={destination} />
+              <span>{truncatedPublicKey(destination)}</span>
+            </div>
+          )}
```

Comment:
I see that this is very similar to 
```
      {federationAddress ? (
        <div>
          <span>{federationAddress}</span>
        </div>
      ) : (
        <div className="SubmitResult__identicon">
          <IdenticonImg publicKey={destination} />
          <span>{truncatedPublicKey(destination)}</span>
        </div>
      )}
```

We should consolidate this repeated logic into a reusable component. I know the styles aren't exactly the same for each, but there's a good amount of overlap. Looks like it's just adding a border to `SubmitResult`. It'd be nice to reuse this as a I can see needing to use this whenever we're dealing with federation addresses.

Reply:
good point, moved it to basics with a new sendPayment folder https://github.com/stellar/freighter/pull/409/commits/5f85e7682491eb14d2ed2f0cf955da1417c5e0b2